File: Hospital.Server/Entities/Response/DispenseItemResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class DispenseItemResponse
    {
        public long Id { get; set; }
        public long DispenseId { get; set; }
        public long MedicineId { get; set; }
        public long? PrescriptionItemId { get; set; }
        public string OriginalMedicineName { get; set; } = string.Empty;
        public string DispensedMedicineName { get; set; } = string.Empty;
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public bool WasSubstituted { get; set; }
        public string? SubstitutionReason { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }
    }
}

File: hospital.client/src/types/DispenseItemResponse.ts
export interface DispenseItemResponse {
  id: number;
  dispenseId: number;
  medicineId: number;
  prescriptionItemId?: number | null;
  originalMedicineName: string;
  dispensedMedicineName: string;
  quantity: number;
  unitPrice?: number | null;
  wasSubstituted?: boolean | null;
  substitutionReason?: string | null;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;
}

export interface DispenseItemRequest {
  id?: number | null;
  dispenseId?: number | null;
  medicineId?: number | null;
  prescriptionItemId?: number | null;
  originalMedicineName?: string | null;
  dispensedMedicineName?: string | null;
  quantity?: number | null;
  unitPrice?: number | null;
  wasSubstituted?: boolean | null;
  substitutionReason?: string | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}
